stdlib/src/uuid.rs

```diff
@@ -0,0 +1,53 @@
+pub(crate) use _uuid::make_module;
+
+#[pymodule]
+mod _uuid {
+
+    use crate::{builtins::PyNone, vm::VirtualMachine};
+    use mac_address::get_mac_address;
+    use once_cell::sync::OnceCell;
+    use rand::Rng;
+    use std::time::{Duration, SystemTime};
+    use uuid::{
+        v1::{Context, Timestamp},
+        Uuid,
+    };
+
+    static NODE_ID: OnceCell<[u8; 6]> = OnceCell::new();
+
+    fn get_node_id() -> &'static [u8; 6] {
+        match get_mac_address() {
+            Ok(Some(_ma)) => {
+                let node_id = NODE_ID.get_or_init(|| get_mac_address().unwrap().unwrap().bytes());
+                node_id
+            }
+            Ok(None) => {
+                let node_id = NODE_ID.get_or_init(|| rand::thread_rng().gen::<[u8; 6]>());
+                node_id
+            }
+            Err(_e) => {
+                let node_id = NODE_ID.get_or_init(|| rand::thread_rng().gen::<[u8; 6]>());
+                node_id
+            }
+        }
```

Comment:
this is checking `get_mac_address` every time. That probably is not expectation for node id. `NODE_ID.get_or_init` must be the outmost wrapper of this function